docs/FIELD_ACCESS.md

```diff
@@ -0,0 +1,456 @@
+| Amount | 393217 | Amount | 8 |
+| Condition | 458769 | Blob | 32 |
+| CancelAfter | 262156 | UInt32 | 4 |
+| FinishAfter | 262157 | UInt32 | 4 |
+| SourceTag | 196624 | UInt32 | 4 |
+| DestinationTag | 196622 | UInt32 | 4 |
+
+### Array Fields
+
+| Field | Code | Type |
+|-------|------|------|
+| Memos | 983049 | Array of Memo |
+| Signers | 983043 | Array of Signer |
+| SignerEntries | 720900 | Array of SignerEntry |
+| Paths | 65537 | Array of Path |
+
+### Nested Object Fields
+
+| Parent | Field | Code | Type |
+|--------|-------|------|------|
+| Memo | MemoType | 458764 | Blob |
+| Memo | MemoData | 458765 | Blob |
+| Memo | MemoFormat | 458766 | Blob |
+| Signer | Account | 524291 | AccountID |
+| Signer | TxnSignature | 524292 | Blob |
+| Signer | SigningPubKey | 524293 | Blob |
+
+## Troubleshooting
+
+### Common Errors
+
+#### FieldNotFound (-2)
+- Field doesn't exist in the object
+- Optional field not present
+- Wrong field code used
+
+**Solution**: Check if field is optional, verify field code
+
+#### NotLeafField (-5)
+- Trying to read an object/array as a value
+- Missing array index in locator
+
+**Solution**: Add array index or access sub-fields
+
+#### LocatorMalformed (-6)
+- Locator exceeds 64 bytes
+- Invalid packing sequence
+- Corrupted locator data
+
+**Solution**: Check locator depth, verify packing order
+
+#### NoArray (-4)
+- Expected array field not found
+- Wrong field code for array
+
+**Solution**: Verify field is actually an array type
+
+### Debugging Tips
+
+1. **Use trace functions** to output field values during development
+2. **Check array lengths** before accessing elements
+3. **Verify field existence** with proper error handling
+4. **Test with fixtures** that have all optional fields
+5. **Compare with JSON** to understand structure (but follow internal paths!)
+
+### Best Practices
+
+1. **Cache array lengths** to avoid repeated calls
+2. **Reuse locators** when accessing multiple fields at same level
+3. **Handle optional fields** gracefully
+4. **Use constants** for field codes
+5. **Document** expected fields
+
+## Examples
+
+### Complete Memo Processing
+
+```rust
+use xrpl_std::locator::Locator;
+use xrpl_std::host::{get_tx_array_len, get_tx_nested_field};
+use xrpl_std::sfield;
+
+fn process_memos() -> Result<()> {
+    // Get memo count
+    let count = get_tx_array_len(sfield::Memos)?;
+    
+    for i in 0..count {
+        // Build locator for MemoType
+        let mut type_loc = Locator::new();
+        type_loc.pack(sfield::Memos);
+        type_loc.pack(i);
+        type_loc.pack(sfield::MemoType);
+        
+        // Read MemoType
+        let mut type_buf = [0u8; 256];
+        let type_len = get_tx_nested_field(
+            type_loc.buffer.as_ptr(),
+            type_loc.cur_buffer_index,
+            type_buf.as_mut_ptr(),
+            type_buf.len()
+        )?;
+        
+        // Build locator for MemoData
+        let mut data_loc = Locator::new();
```

Comment:
`repack_last()` function can be used. It replaces the last field of the locator.